ipld/resolver/src/service.rs

```diff
@@ -672,46 +674,77 @@ pub fn build_transport(local_key: Keypair) -> Boxed<(PeerId, StreamMuxerBox)> {
 
 async fn download_blob(
     iroh: Iroh,
-    hash: Hash,
+    seq_hash: Hash,
     size: u64,
     node_addr: NodeAddr,
 ) -> anyhow::Result<()> {
+    // Download top-level blob
     // Use an explicit tag so we can keep track of it
-    // TODO: this needs to be tagged with a "user id"
-    let tag = iroh::blobs::Tag(format!("stored-seq-{hash}").into());
-    let res = iroh
-        .blobs()
+    let tag = iroh::blobs::Tag(format!("stored-seq-{seq_hash}").into());
+    iroh.blobs()
         .download_with_opts(
-            hash,
+            seq_hash,
             iroh::client::blobs::DownloadOptions {
                 format: iroh::blobs::BlobFormat::Raw,
-                nodes: vec![node_addr],
+                nodes: vec![node_addr.clone()],
                 tag: iroh::blobs::util::SetTagOption::Named(tag),
                 mode: iroh::client::blobs::DownloadMode::Queued,
             },
         )
         .await?
         .await?;
 
-    if res.local_size + res.downloaded_size != size {
-        return Err(anyhow!(
-            "downloaded blob size {} does not match expected size {}",
-            res.local_size + res.downloaded_size,
-            size
-        ));
+    // Read top-level blob as a hash sequence
+    let res = iroh
+        .blobs()
+        .read_to_bytes(seq_hash)
+        .await
+        .map_err(|e| anyhow!("failed to read hash sequence object: {} {}", seq_hash, e))?;
+    let hash_seq = HashSeq::try_from(res)
+        .map_err(|e| anyhow!("failed to parse hash sequence object: {} {}", seq_hash, e))?;
+
+    // Download blobs in the hash sequence
```

Comment:
@dignifiedquire seems this is required, ie, downloading just the top level blob does not make the hashes in the sequence available.

Reply:
see my comment above, this loop is not needed when adjusting the download of the hashseq itself